test/test_datapipe.py

```diff
@@ -907,6 +907,25 @@ def test_picklable(self):
                 with self.assertRaises(AttributeError):
                     p = pickle.dumps(datapipe)
 
+    def test_concat_datapipe(self):
+        input_dp1 = MDP(range(10))
+        input_dp2 = MDP(range(5))
+
+        with self.assertRaisesRegex(ValueError, r"Expected at least one DataPipe"):
+            dp.map.Concat()
+
+        with self.assertRaisesRegex(TypeError, r"Expected all inputs to be `MapDataPipe`"):
+            dp.map.Concat(input_dp1, ())  # type: ignore[arg-type]
+
+        concat_dp = input_dp1.concat(input_dp2)
+        self.assertEqual(len(concat_dp), 15)
+        for index in range(15):
+            self.assertEqual(concat_dp[index], (list(range(10)) + list(range(5)))[index])
```

Comment:
Just wondering does `list(concat_dp)` work?

Reply:
Nice catch! I have changed the error in `__getitem__` from `ValueError` to `IndexError` so that `list(concat_dp)` works now. I've also added a test on this.